hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/CompressionContext.java

```diff
@@ -34,21 +36,49 @@
 @InterfaceAudience.LimitedPrivate({HBaseInterfaceAudience.COPROC, HBaseInterfaceAudience.PHOENIX})
 public class CompressionContext {
 
-  static final String ENABLE_WAL_TAGS_COMPRESSION =
+  public static final String ENABLE_WAL_TAGS_COMPRESSION =
       "hbase.regionserver.wal.tags.enablecompression";
 
+  public static final String ENABLE_WAL_VALUE_COMPRESSION =
+      "hbase.regionserver.wal.value.enablecompression";
+
   public enum DictionaryIndex {
     REGION, TABLE, FAMILY, QUALIFIER, ROW
   }
 
+  static class ValueCompressor {
+    final Deflater deflater;
+    final Inflater inflater;
+
+    public ValueCompressor() {
+      deflater = new Deflater();
+      inflater = new Inflater();
```

Comment:
All the inflate deflate methods are synchronized on a single object, for example,  Is that a a concern for bottleneck since we are sharing instances of them, just curious.

```
>   public int inflate(byte[] b, int off, int len)
>         throws DataFormatException
>     {
>         if (b == null) {
>             throw new NullPointerException();
>         }
>         if (off < 0 || len < 0 || off > b.length - len) {
>             throw new ArrayIndexOutOfBoundsException();
>         }
>         synchronized (zsRef) {  <===
 >             ensureOpen();
>             int thisLen = this.len;
>             int n = inflateBytes(zsRef.address(), b, off, len);
>             bytesWritten += n;
>             bytesRead += (thisLen - this.len);
>             return n;
>         }
>     }
```

Reply:
There is no sharing. One Compressor instance per WAL writer.

Reply:
The append is single threaded, so I do not think the synchronized itself would cause much overhead.

But for performance, on modern hardwares, with powerful NIC and SSD, now the bottleneck may become CPU so enable compression itself may impact performance.

But anyway, you could choose not enable it if the bottleneck on your cluster is CPU so it is not a blocker for adding compression support.